services/blockchain-connector/shared/sdk/events.js

```diff
@@ -68,8 +75,19 @@ const getEventsByHeightFormatted = async (height) => {
 	return formattedEvents;
 };
 
+const verifyClientConnection = async () => {
```

Comment:
I'd move all this logic to `client.js` and use Signals to communicate the receipt of events over the API client.